examples/example_app/example_app/jobs.py

```diff
@@ -3,23 +3,271 @@
+        default={},
+    )
+    # the name `dryrun` is significant!
+    dryrun = DryRunVar(description="Set to True to run in dry-run mode, bypassing approval requirements")
+
+    def before_start(self, task_id, args, kwargs):
+        """
+        Called before starting the Job run() method.
+
+        If any unhandled exception is raised, run() will not be called and instead the Job will proceed directly to
+        calling on_failure().
+
+        Args:
+            task_id (str): Present for Celery compatibility, can be ignored in most cases.
+            args (list): Present for Celery compatibility, can be ignored in most cases.
+            kwargs (dict): The keyword args that will be passed into `run()`.
+        """
+        self.logger.info("Before start! The provided kwargs are `%s`", kwargs)
+
+    def run(self, **kwargs):
+        """
+        The main worker function of any Job.
+
+        Considered to have "succeeded" if it returns without raising any exception; any raised and unhandled exception
+        will be considered a "failure".
+
+        Args:
+            **kwargs (dict): Arguments corresponding to the ScriptVariables defined for this Job. Be sure to define
+                default values for any variables that are marked as optional!
+
+        Returns:
+            data (any): Data that will be saved to the JobResult. **Must** be serializable to JSON.
+        """
+        self.logger.info("Running!")
+        for key, value in kwargs.items():
+            self.logger.info("For kwarg %s, the provided value was `%s` `%s`", key, type(value), value)
+
+        # A few relevant instance attributes are automatically set when a Job runs:
+        self.logger.debug("The user who requested this Job to run is %s", self.user.username)
+        self.logger.debug("The JobResult for this Job has id %s", self.job_result.id)
+
+        # Since Jobs are Python classes, you can define any custom methods you like and call them yourself:
+        self.demonstrate_logging(kwargs)
+
+        # The create_file() helper method takes a filename and either a string or bytestring as input.
+        # Users can subsequently download the created file.
+        self.create_file("example.txt", "\n".join([kwargs["string_input"], kwargs["text_input"]]))
+
+        if kwargs["should_fail"]:
+            raise RuntimeError("This unhandled exception will cause the Job to fail")
+        # else:
+        return {"my job result": {"The return value on success can contain any": ["JSON", "serializable", "data"]}}
+
+    def demonstrate_logging(self, kwargs):
+        """Not a magic method - this is a custom Python function manually called from run()."""
+        self.logger.debug("This is a simple debug message, with **Markdown** formatting.")
+        self.logger.info(
+            "This is an info message, with an associated database object",
+            extra={"object": kwargs["location_type_input"]},
+        )
```

Comment:
Let's add a `logger.success` case to the ExampleEverythingJob now that that's an option in 2.4.